pwiz_tools/Skyline/TestFunctional/AreaCVHistogramTest.cs

```diff
@@ -377,10 +377,13 @@ private void TestRefinement()
             var refineDlg = ShowDialog<RefineDlg>(() => SkylineWindow.ShowRefineDlg());
             RunUI(() => { refineDlg.CVCutoff = 20; });
             OkDialog(refineDlg, refineDlg.OkDialog);
-            var doc = SkylineWindow.Document;
-            var refineDocState = (doc.PeptideGroupCount, doc.PeptideCount, doc.PeptideTransitionGroupCount,
-                doc.PeptideTransitionCount);
-            Assert.AreEqual(graphStates[0], refineDocState);
+            WaitForCondition(() =>

```

Comment:
If you wanted to, a different way to check whether SkylineWindow.ShowRefineDlg has finished its work is to do a "WaitForDocumentChange".

But, this way works too.